rascal-vscode-extension/src/lsp/ParameterizedLanguageServer.ts

```diff
@@ -152,8 +152,8 @@ export interface LanguageParameter {
     pathConfig: string
     /** name of the language */
     name: string;
-    /** extension for files in this language */
-    extension:string;
```

Comment:
Consider leaving the original singular property and supporting it for backwards compatibility.

Reply:
yup, I was also thinking about that. that would be nicer, as we did the same for the rascal consumption of the API.

Reply:
The problem is that the new definition would still be an compile time error for consumers, as they would not have defined `extensions` in their object. and I didn't want to make that one optional.

Reply:
Interesting alternatives! I understand this discussion, but are we sure we have to be backward compatible on this side? In principle, our own server side (in Java) is the only client of this interface, and supplier of this data (as far as I know). And the more nice tricks we do here in TypeScript, the more nasty this object will look in Java. The java JSON-RPC bridge does something "special" for union types, but that's as far as my compliments go ;-)

Reply:
I don't know how much this change will affect consumers of [this package](https://www.npmjs.com/package/@usethesource/rascal-vscode-dsl-lsp-server). Maybe it is fine to break the few (?) consumers. If that's true, no problem. It has been "considered" and rejected 😄

Reply:
I think the concern is that this LanguageParameter is also used by deployed languages. 

I'm happy breaking that API a bit, since we are in 0.x, and the compile error would be very present in node, and only happen if you explicitly upgrade that dependency.

At the moment, I know all the places where this is happening.

Reply:
Ok, we are argee :)

Reply:
nice to talk about this stuff @urbanfly ! thanks